types/part_set.go

```diff
@@ -271,6 +271,20 @@ func (ps *PartSet) Total() uint32 {
 }
 
 func (ps *PartSet) AddPart(part *Part) (bool, error) {
+	// The proof should be compatible with the number of parts.
+	if part.Proof.Total != int64(ps.total) {
+		return false, ErrPartSetInvalidProof
+	}
+
+	// Check hash proof
+	if part.Proof.Verify(ps.Hash(), part.Bytes) != nil {
+		return false, ErrPartSetInvalidProof
+	}
+
+	return ps.AddPartWithoutProof(part)
+}
+
+func (ps *PartSet) AddPartWithoutProof(part *Part) (bool, error) {
```

Comment:
I don't see where else we call this. What was the reason for splitting?

Reply:
it's called in Celestia-node to have the parts and build the block without verifying the proof

Reply:
why do we want to build the block without verifying the proof

Reply:
Since they make the assumption that every bridge node will be connected to a local full node. So the data is trusted and no need to verify inclusion. Also to reduce the size of the parts.

The initial implementation provided the proofs by default, but they said its better to disable it by default

Reply:
where there was the check?

I guess it's good to add a check anyway. good catch 👍

Reply:
8bbbe85571e3546035fdc42d1f76990237b1dca6

Reply:
8bbbe85571e3546035fdc42d1f76990237b1dca6